server/service/orbit.go

```diff
@@ -181,6 +182,33 @@ func (svc *Service) GetOrbitConfig(ctx context.Context) (fleet.OrbitConfig, erro
 		return fleet.OrbitConfig{}, err
 	}
 
+	var hostOS *fleet.OperatingSystem
+	if host.ID != 0 {
+		hostOS, err = svc.ds.GetHostOperatingSystem(ctx, host.ID)
+		if err != nil {
+			return fleet.OrbitConfig{}, ctxerr.Wrap(ctx, err, "get orbit config")
+		}
+	} else {
+		hostOS = &fleet.OperatingSystem{}
+	}
+
+	var macOsVersionFloat float32
+	if hostOS.Platform == "darwin" {
```

Comment:
given that this endpoint gets called every 30 seconds for every host, I'm a bit concerned about load (we did see a spike in DB resources when we added scripts here.)

I have two ideas, you might have others:

1. Call `svc.ds.GetHostOperatingSystem` only if:
      1. `host.Platform == "darwin"` (you already have the platform in the `host` struct)
      2. macOS is enabled and configured 
      3. macOS updates are enabled for the host
      4. any other thing you might see
5. In `LoadHostByOrbitNodeKey`, `JOIN` with `host_operating_system` and pick `version`

Reply:
Very true, I'll look into making those changes

Reply:
👍 sounds good to me, we can adjust as we see fit